uwblueprint/sistema
# Seeding script

# Notion ticket link


[Create Seeding Script](https://www.notion.so/uwblueprintexecs/Create-Seeding-Script-84d4983a336e4db6b6dda03d8b77d4dc?pvs=4)



## Implementation description
- Added the seeding script using snaplet to implement seeding and used faker to generate date
- Added the seeding script running, sync and migration database commands to package.json
 



## Steps to test

It will automatically seed when you run `docker compose up --build`

### Steps to test manually*

1. Ensure the database is running locally and make sure you have initialized the database `npx prisma migrate dev --name init`
2. In the root directory,  `npx @snaplet/seed sync` to sync the seed file to any change in the data models of the database
3. `npx prisma db seed` to seed the database based on seed.ts
4. Run `docker exec -it sistema-db-1 /bin/bash` and then in the container run `psql -U sistema -d sistema` to enter the Postgres shell
6. In the Postgres shell check if the database tables are seeded by running `SELECT * FROM public."<Table-name>";` eg: `SELECT * FROM public."Absence";`



## What should reviewers focus on?
- All the tables are seeded correctly with correct data and are in sync with each other

## Checklist
